### ISSUE TITLE
CKV_AWS_111 not failing with `arn:aws:s3:::*/*`

### ISSUE THREAD

**Issue by caevv** [bridgecrewio/checkov#5718]
The following CDK code:

```js
const writeAnyS3PolicyStatement = new aws_iam.PolicyStatement({
  actions: ['s3:PutObject', 's3:PutObjectAcl'],
  resources: ['arn:aws:s3:::*/*'],
});

const role = new aws_iam.Role(this, 'RandomRole', {
  assumedBy: new aws_iam.ServicePrincipal('lambda.amazonaws.com')
});

role.addToPolicy(writeAnyS3PolicyStatement);
```

Produces this CloudFormation template:

```json
"RandomRoleDefaultPolicy930364B8": {
 "Type": "AWS::IAM::Policy",
 "Properties": {
  "PolicyDocument": {
   "Statement": [
    {
     "Action": [
      "s3:PutObject",
      "s3:PutObjectAcl"
     ],
     "Effect": "Allow",
     "Resource": "arn:aws:s3:::*/*"
    }
   ],
   "Version": "2012-10-17"
  },
  "PolicyName": "RandomRoleDefaultPolicy930364B8",
  "Roles": [
   {
    "Ref": "RandomRole17A9849E"
   }
  ]
 },
 "Metadata": {
  "aws:cdk:path": "CdkTsSampleAppStack-test/RandomRole/DefaultPolicy/Resource"
 }
},
```

But it doesn't detect this as permissive permission. Why?

```bash
❯ docker run --tty --volume $(PWD):/tf --workdir /tf bridgecrew/checkov --directory /tf

Check: CKV_AWS_111: "Ensure IAM policies does not allow write access without constraints"
	PASSED for resource: AWS::IAM::Policy.RandomRoleDefaultPolicy930364B8
	File: /cdk.out/test.template.json:366-392
	Guide: https://docs.prismacloud.io/en/enterprise-edition/policy-reference/aws-policies/aws-iam-policies/ensure-iam-policies-do-not-allow-write-access-without-constraint
```

**Version (please complete the following information):**
3.0.14

**Additional context**
```
❯ cat .checkov.yaml
skip-check: CKV_AWS_117,CKV_AWS_116,CKV_AWS_115,CKV_AWS_158
```

**Comment by gruebel** [comment]
hey @caevv thanks for reaching out.

The current logic looks for unrestricted wildcard usage. If a resource targets a specific service like in your example, then it will already pass the check.

**Comment by caevv** [comment]
Thanks for getting back to me @gruebel.

So it can write to any S3? Looks pretty permissive. Ideally it would write to a specific bucket.

**Comment by caevv** [comment]
I've tried a different role and same issue:

```ts
const bucket = new S3Bucket(this, "Bucket", {
  bucketName: "some-bucket",
  removalPolicy: RemovalPolicy.DESTROY,
  autoDeleteObjects: true,
});

const writeAnyS3PolicyStatement = new aws_iam.PolicyStatement({
  sid: "AllowBucketActions",
  effect: aws_iam.Effect.ALLOW,
  actions: ["s3:PutBucketPolicy", "s3:GetBucketLocation", "s3:ListBucket", "s3:GetObject"],
  resources: [`arn:aws:s3:::${bucket.bucketName}`, `arn:aws:s3:::${bucket.bucketName}/*`],
});

const role = new aws_iam.Role(this, "RandomRole", {
  assumedBy: new aws_iam.ServicePrincipal("lambda.amazonaws.com"),
});

role.addToPolicy(writeAnyS3PolicyStatement);
```



CloudFormation generated:

```json
"DefaultPolicy": {
  "id": "DefaultPolicy",
  "path": "CdkTsSampleAppStack-test/RandomRole/DefaultPolicy",
  "children": {
    "Resource": {
      "id": "Resource",
      "path": "CdkTsSampleAppStack-test/RandomRole/DefaultPolicy/Resource",
      "attributes": {
        "aws:cdk:cloudformation:type": "AWS::IAM::Policy",
        "aws:cdk:cloudformation:props": {
          "policyDocument": {
            "Statement": [
              {
                "Action": [
                  "s3:GetBucketLocation",
                  "s3:GetObject",
                  "s3:ListBucket",
                  "s3:PutBucketPolicy"
                ],
                "Effect": "Allow",
                "Resource": [
                  {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:s3:::",
                        {
                          "Ref": "Bucket83908E77"
                        },
                        "/*"
                      ]
                    ]
                  },
                  {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:s3:::",
                        {
                          "Ref": "Bucket83908E77"
                        }
                      ]
                    ]
                  }
                ],
                "Sid": "AllowBucketActions"
              }
            ],
            "Version": "2012-10-17"
          },
          "policyName": "RandomRoleDefaultPolicy930364B8",
          "roles": [
            {
              "Ref": "RandomRole17A9849E"
            }
          ]
        }
      },
      "constructInfo": {
        "fqn": "aws-cdk-lib.aws_iam.CfnPolicy",
        "version": "2.78.0"
      }
    }
  },
  "constructInfo": {
    "fqn": "aws-cdk-lib.aws_iam.Policy",
    "version": "2.78.0"
  }
}
```

**Comment by ArjunMenon-bit** [comment]
@gruebel  Can you guide me a little on exactly where this logic is written. Would be more than happy to contribute for a fix to this issue.

**Comment by JamesWoolfenden** [comment]
@ArjunMenon-bit the file in question is here. The check logic is inherited from cloudsplaining
